plugins/sqlite_authkey_storage_plugin/src/main/kotlin/de/busesteinkamp/plugins/auth/SqliteAuthKeyRepository.kt

```diff
@@ -0,0 +1,77 @@
+package de.busesteinkamp.plugins.auth
+
+import de.busesteinkamp.domain.auth.AuthKey
+import de.busesteinkamp.domain.auth.AuthKeyRepository
+import java.sql.DriverManager
+import java.text.SimpleDateFormat
+import java.util.*
+
+class SqliteAuthKeyRepository : AuthKeyRepository {
+    private val url = "jdbc:sqlite:auth_keys.db"
+
+    init {
+        Class.forName("org.sqlite.JDBC")
+
+        val connection = DriverManager.getConnection(url)
+        val statement = connection.createStatement()
+        statement.executeUpdate(
+            "CREATE TABLE IF NOT EXISTS auth_keys (platformName TEXT, authKey TEXT, createdAt TEXT, expiresAt TEXT)"
+        )
+        statement.close()
+        connection.close()
+    }
+
+    override fun find(platformName: String): AuthKey? {
+        val connection = DriverManager.getConnection(url)
+        val statement = connection.createStatement()
+        val resultSet = statement.executeQuery(
+            "SELECT authKey, createdAt, expiresAt FROM auth_keys WHERE platformName = '$platformName'"
+        )
+        val authKey = if (resultSet.next()) {
+            AuthKey(platformName, resultSet.getString("authKey"), parseDate(resultSet.getString("createdAt")), parseDate(resultSet.getString("expiresAt")))
+        } else {
+            null
+        }
+        resultSet.close()
+        statement.close()
+        connection.close()
+        return authKey
+    }
```

Comment:
Die Funktionen öffnen Tür und Tor für SQL Injections, und Fehlerbehandlung der Connection ist auch nicht optimal. Hier ist eine Änderung, die theoretisch für alle Übernommen werden kann (natürlich angepasst)

```suggestion
    override fun find(platformName: String): AuthKey? {
        val sql = "SELECT authKey, createdAt, expiresAt FROM auth_keys WHERE platformName = ?"
        try (val connection = DriverManager.getConnection(url);
             val preparedStatement = connection.prepareStatement(sql)) {
            preparedStatement.setString(1, platformName)
            val resultSet = preparedStatement.executeQuery()
            return if (resultSet.next()) {
                AuthKey(platformName, resultSet.getString("authKey"), parseDate(resultSet.getString("createdAt")), parseDate(resultSet.getString("expiresAt")))
            } else {
                null
            }
        } catch (e: SQLException) {
            throw RuntimeException("Error finding auth key", e)
        }
    }
```